ui/src/app/features/face-recognition/recognition-result/recognition-result.component.ts

```diff
@@ -102,20 +105,61 @@ export class RecognitionResultComponent implements OnChanges, OnDestroy {
     ctx.fillText(face.face_name, box.x_min + 10, box.y_min - 5);
   }
 
+  private createDetectionImage(ctx, box) {
+    ctx.beginPath();
+    ctx.strokeStyle = 'green';
+    ctx.moveTo(box.x_min, box.y_min);
+    ctx.lineTo(box.x_max, box.y_min);
+    ctx.lineTo(box.x_max, box.y_max);
+    ctx.lineTo(box.x_min, box.y_max);
+    ctx.lineTo(box.x_min, box.y_min);
+    ctx.stroke();
+    ctx.fillStyle = 'green';
+    ctx.fillRect(box.x_min, box.y_max, box.x_max - box.x_min, this.faceDescriptionHeight);
+    ctx.fillStyle = 'white';
+    ctx.font = '12pt Roboto Regular Helvetica Neue sans-serif';
+    ctx.fillText(box.probability.toFixed(4), box.x_min + 10, box.y_max + 20);
+  }
+
   /*
    * Make canvas and draw face and info on image.
    *
    * @preparedData prepared box data and faces.
    */
   drawCanvas(preparedData) {
+    switch (this.serviceType) {
+      case ServiceTypes.Recognition:
+        this.drawRecognitionCanvas(preparedData);
+        break;
+
+      case ServiceTypes.Detection:
+        this.drawDetectionCanvas(preparedData);
+        break;
+    }
+  }
+
+  drawRecognitionCanvas(data) {
     const img = new Image();
     const ctx: CanvasRenderingContext2D = this.myCanvas.nativeElement.getContext('2d');
     img.onload = () => {
       ctx.drawImage(img, 0, 0, this.canvasSize.width, this.canvasSize.height);
-      for (const value of preparedData) {
+      for (const value of data) {
         // eslint-disable-next-line @typescript-eslint/naming-convention
         const resultFace = value.faces.length > 0 ? value.faces[0] : { face_name: undefined, similarity: 0 };
-        this.createImage(ctx, value.box, resultFace);
+        this.createRecognitionImage(ctx, value.box, resultFace);
+      }
+    };
+    img.src = URL.createObjectURL(this.file);
+  }
+
+  drawDetectionCanvas(data) {
```

Comment:
same as above very similar to drawRecognitionCanvas, on ly code inside for (const value of data) { is different